### DISCUSSION TITLE
Improve file upload (e.g. drag and drop)?

### DISCUSSION THREAD

**Discussion by Glenno-H** [zauberzeug/nicegui#74] [Ideas / Feature Requests]
Is it possible to have a Drag and Drop option ?  Would be great to be able to drop something like a CSV and be able to process it.

**Comment by falkoschindler** [comment]
The `ui.upload` element already supports dropping files. See #75 for an example.

I'm also thinking about replacing the `ui.upload` element with Quasar's nicer [File Picker](https://quasar.dev/vue-components/file-picker) or [Uploader](https://quasar.dev/vue-components/uploader). But this seems to require some changes in JustPy, which is currently undergoing quite some refactoring. Therefore I'll wait until this is settled before attempting to add more features that depend on JustPy.

**Reply by Glenno-H** [reply]
Thanks very much, Ill play around a bit more and see if I can get the upload element to do what I need.  You are right, the Quasar stuff looks very nice!   Thanks for your efforts!

**Comment by falkoschindler** [comment]
We just released NiceGUI with an improved `ui.upload` which also supports drag & drop:
https://nicegui.io/reference#file_upload
